### PR TITLE
More py3

### PR DESCRIPTION
**Pull request by bcail**

### CODE REVIEW

**Comment by bcail** [comment]
@alexwlchan this PR includes the commit from PR #399.

File: loris/authorizer.py
Comment (alexwlchan) [inline loris/authorizer.py:56]: Does this line not work unmodified on Python 3?

**Edit:** Wait, I see – we're mutating the dict. Never mind.

File: loris/authorizer.py
Comment (alexwlchan) [inline loris/authorizer.py]: I think we should either drop a warning log that we're encoding the `salt` config option, so there's an explanation if for some reason we've guessed the wrong encoding; or just not encode the string at all, and throw a TypeError instructing the user to change their config.

File: loris/authorizer.py
Comment (alexwlchan) [inline loris/authorizer.py]: Ditto above.

File: loris/authorizer.py
Comment (alexwlchan) [inline loris/authorizer.py]: Rather than encoding it in multiple places, we should do the encode once, at the top of the function.

File: loris/authorizer.py
Comment (alexwlchan) [inline loris/authorizer.py]: EAFP suggests it might be preferable to write:

```python
try:
    value.decode('utf8')
except AttributeError:
    pass
```

but I don't think it makes a huge amount of difference.

**Comment by bcail** [comment]
@alexwlchan thanks for your review. I've made updates.

File: loris/authorizer.py
Comment (alexwlchan) [inline loris/authorizer.py]: nit, while we're editing this line: logging calls should use commas for string substitution, i.e.

```python
logger.debug('Got basic origin: %s', origin)
```

This avoids doing the string substitution if the log line is never used.

File: loris/authorizer.py
Comment (alexwlchan) [reply loris/authorizer.py:56]: Mmm. I think you can iterate over `list(svc.items())` in the same way as before.

Or might it be even simpler just to replace this with a dict comprehension? i.e.

```python
return {k: v for k, v in svc.items() if v}
```

You no longer get in-place modifications, but it feels weird for the function to edit in-place *and* return. It should do one or the other.

File: loris/authorizer.py
Comment (alexwlchan) [inline loris/authorizer.py]: Let's include the problem value for easy debugging. Something like:

```python
raise ConfigError(
    '"salt" config parameter must be bytes; got %r (%s)' % (config['salt'], type(config))
)
```

**Comment by bcail** [comment]
@alexwlchan are there any more changes you want me to make before this PR is merged?

**Comment by alexwlchan** [comment]
I’ve been sitting on a half-complete review for days – I’ll try to finish it today or tomorrow.

File: tests/webapp_t.py
Comment (alexwlchan) [inline tests/webapp_t.py:281]: Suggestion: rather than decoding `resp.data`, it might be cleaner to do the comparison with a bytestring:

```python
self.assertTrue(resp.data.startswith(b'This is Loris, '))
```

Ditto all the tests below.

**Review by alexwlchan — APPROVED** [review]
Okay, a couple of suggestions, but nothing too major. The improved ConfigError is the only must-address; everything else I'll leave up to you.

Happy for you to merge once you've made the markups. I don't need another review.

**Pull request merged**
